src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ImperialFirePatch.cs

```diff
@@ -107,7 +107,12 @@ public override void Reset()
     private static bool AnyPartyMemberHasThePerkActive(MobileParty party) {
       var perk = ActivePatch._perk;
       var partyMemberValue = new ExplainedNumber(0f);
-      PerkHelper.AddPerkBonusForParty(perk, Hero.MainHero.PartyBelongedTo, ref partyMemberValue);
+      PerkHelper.AddPerkBonusForParty(perk, party, ref partyMemberValue);
+
+      if (party.Army?.Parties != null)
+        foreach (var armyParty in party.Army?.Parties)
+          PerkHelper.AddPerkBonusForParty(perk, armyParty, ref partyMemberValue);
```

Comment:
The function `AddPerkBonusForParty` is being called twice for the initial party, once with `party`, and once inside the foreach loop.

Reply:
Since the perk is a enable/disable feature, I think that is ok, or no?

Reply:
I don't think so, since the enable/disable condition is `partyMemberValue.ResultNumber.IsEqualOrBiggerThan(1f);`, if you add the same factor twice it will affect the result.

Reply:
Ah, I see it now, you changed the Perk primaryValue to 1. I guess it doesn't make a difference then, but if you don't need the primaryValue and just a boolean, I think it would be much clearer if you wrote
```
if (party.Army == null) 
    return party.LeaderHero.GetPerkValue(__perk);
foreach (var armyParty in party.Army.Parties) {
    if (armyParty.LeaderHero.GetPerkValue(__perk)
        return true;
return false;
```
instead of taking using `PerkHelper` to add a bunch of 1s.

Reply:
I believe the code you wrote don't check the party members of the clans. I just used the PerkHelper to make that job easier.

Reply:
I see. Well, it's fine.